From 09ae9247749aa7768655696fd099561322b1a90e Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:09:42 +0200
Subject: [PATCH] tests: shell: skip ip option tests if kernel does not support
 it

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit c624578ee18131492e7f72b662d5faf7c042e0d8

commit c624578ee18131492e7f72b662d5faf7c042e0d8
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Thu Jun 13 00:37:48 2024 +0200

    tests: shell: skip ip option tests if kernel does not support it

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 tests/shell/features/ip_options.nft      | 8 ++++++++
 tests/shell/testcases/sets/typeof_sets_0 | 2 ++
 2 files changed, 10 insertions(+)
 create mode 100644 tests/shell/features/ip_options.nft

diff --git a/tests/shell/features/ip_options.nft b/tests/shell/features/ip_options.nft
new file mode 100644
index 0000000..0b8cb09
--- /dev/null
+++ b/tests/shell/features/ip_options.nft
@@ -0,0 +1,8 @@
+# dbb5281a1f84 ("netfilter: nf_tables: add support for matching IPv4 options")
+# v5.3-rc1~140^2~153^2~1
+
+table ip x {
+	chain y {
+		ip option ra value 255
+	}
+}
diff --git a/tests/shell/testcases/sets/typeof_sets_0 b/tests/shell/testcases/sets/typeof_sets_0
index 943c9c2..3a4ed94 100755
--- a/tests/shell/testcases/sets/typeof_sets_0
+++ b/tests/shell/testcases/sets/typeof_sets_0
@@ -4,6 +4,8 @@
 # s1 and s2 are identical, they just use different
 # ways for declaration.
 
+# NFT_TEST_REQUIRES(NFT_TEST_HAVE_ip_options)
+
 set -e
 
 die() {
